packages/starksheet-webapp/src/provider/starknetProvider.ts

```diff
@@ -21,26 +25,41 @@ export class StarknetProvider implements ChainProvider {
     this.rpcProvider.getChainId().then((id) => (this.chainId = hex2str(id)));
   }
 
+  /**
+   * @inheritDoc
+   */
   getExplorerAddress(contractAddress: string) {
     return this.chainId === "SN_MAIN"
       ? `https://starkscan.co/contract/${contractAddress}`
       : `https://testnet.starkscan.co/contract/${contractAddress}`;
   }
 
+  /**
+   * @inheritDoc
+   */
   getNftMarketplaceAddress(contractAddress: string) {
     return this.chainId === "SN_MAIN"
       ? `https://starkscan.co/contract/${contractAddress}`
```

Comment:
completely. We should also keep this part in mind when refactoring the chain provider general config. This should be derived from a config object rather than comparing the chain id at runtime to find the correct explorer/marketplace website